### PR TITLE
Add CSP, Expect-CT, and HPKP documentation

### PR DESCRIPTION
**Pull request by ndmanvar**

### CODE REVIEW

File: docs/learn/security-policy-reporting.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+Content-Security-Policy
+-----------------------
+
+The `CSP <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy>`_
+response header allows for control over resources can be loaded.
+
+Configure the `Content-Security-Policy` header with relevant policies and report-uri.
+
+For example, in Python this looks like the following:
```
[the remark this answers is not in the document]
Comment (ndmanvar) [reply docs/learn/security-policy-reporting.rst]: I agree. will make that change.

**Comment by ndmanvar** [comment]
@mattrobenolt Made recommended changes (only showing relevant fields). Could you please review and merge?

File: docs/learn/security-policy-reporting.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Content-Security-Policy
+-----------------------
+
+The `CSP <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy>`_
+response header allows for control over resources can be loaded.
+
+Configure the `Content-Security-Policy` header with relevant policies and report-uri.
+
+For example, in Python this looks like the following::
```
Comment (mattrobenolt) [inline docs/learn/security-policy-reporting.rst]: This still says "in Python".

File: docs/learn/security-policy-reporting.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    Content-Security-Policy: ...; report-uri https://sentry.io/api/<project>/csp-report/?sentry_key=<key>
```
Comment (mattrobenolt) [inline docs/learn/security-policy-reporting.rst]: This uri should also just be `/security/`. `/security/` is to replace all of them under the same URL, and we don't need a specific `/csp-report/` anymore.

**Comment by ndmanvar** [comment]
@mattrobenolt Fixed as per your comment as well as using descriptions (including releases/environments) from settings. Could you please review/merge again?

**Pull request merged**